### ISSUE TITLE
java.lang.UnsatisfiedLinkError: dlopen failed: library "libavutil.so.54" not found

### ISSUE THREAD

**Issue by vxhviet** [wseemann/FFmpegMediaMetadataRetriever#92]
Please @wseemann could you elaborate how to get FFmpegMediaMetadataRetriever:1.0.5 working with Android Studio 1.5.1? 

What I've tried so far:

1. 
Add the following maven dependency to your project's `build.gradle` file:
```
dependencies {  
    compile 'com.github.wseemann:FFmpegMediaMetadataRetriever:1.0.5'  
}
```

result in: 
`java.lang.UnsatisfiedLinkError: dlopen failed: library "libavutil.so.54" not found`

when calling:
`FFmpegMediaMetadataRetriever retriever = new FFmpegMediaMetadataRetriever();`

2.
Add the following maven dependency to your project's `build.gradle` file:
```
dependencies {
    compile 'com.github.wseemann:FFmpegMediaMetadataRetriever:1.0.5'
}
```

Add _all-fmmr.aar_ to project's `libs` folder and change:
```
dependencies {
    compile fileTree(dir: 'libs', include: ['*.aar'])
    ...
}
```
result in:
`Warning:Project app: Only Jar-type local dependencies are supported. Cannot handle: ...\app\libs\all-fmmr.aar`

3.
Same stuffs as above but instead rename _all-fmmr.aar_ to _all-fmmr.jar_ and revert `include: ['*.aar']` back to `include: ['*.jar']` result in:
`Error:duplicate files during packaging of APK ...\app\build\outputs\apk\app-debug-unaligned.apk
Path in archive: AndroidManifest.xml`

4.
Same stuff as above but add this to the `build.gradle`:
````
android{
    packagingOptions {
        exclude 'AndroidManifest.xml'
    }
}
```
and getting `java.lang.UnsatisfiedLinkError: dlopen failed: library "libavutil.so.54" not found` again.

5.
I guess Android Studio still haven't recognize the libs so I try `File>New>New Module...>Import .JAR/.AAR Package>all-fmmr.aar` and then add `all-fmmr` module as module dependency to `app` module and get:
`Error:Execution failed for task ':app:transformClassesWithDexForDebug'.
 com.android.build.api.transform.TransformException: com.android.ide.common.process.ProcessException: org.gradle.process.internal.ExecException: Process 'command 'C:\Program Files\Java\jdk1.7.0_79\bin\java.exe'' finished with non-zero exit value 2`

6.
So I tried [this solution](http://stackoverflow.com/questions/34285406/java-exe-finished-with-non-zero-exit-value-2-in-android-studio?answertab=active#tab-top) and get:
`Error:Execution failed for task ':app:transformClassesWithJarMergingForDebug'.
com.android.build.api.transform.TransformException: java.util.zip.ZipException: duplicate entry: wseemann/media/BuildConfig.class`

I actually did tried `./gradlew clean` after that but I guess you all must be so bored with this stuff until now so I won't bother writing it down.

Long story short, is there a clearer instruction to get this lib working in Android Studio?

**Comment by shaileshmulange** [comment]
Facing same issue, please fix it asap.

**Comment by Tamas-F** [comment]
Same problem here.

**Comment by wseemann** [comment]
Please try version 1.0.6:

```
dependencies {
    compile 'com.github.wseemann:FFmpegMediaMetadataRetriever:1.0.6'
}
```

**Comment by Tamas-F** [comment]
For me it solved the issue, thanks :)

**Comment by shaileshmulange** [comment]
@wseemann  Thanks, it fixed the issue. But https is not working.
I have tried following both urls, http is working fine but https is not working.
https://d4tkdpl7iotvy.cloudfront.net/videos/6/1457375907/video.m3u8
http://d4tkdpl7iotvy.cloudfront.net/videos/6/1457375907/video.m3u8

**Comment by vxhviet** [comment]
@wseemann Thanks man, it's really working with the latest build, fast and accurate I must add.
However I can't really use your lib as it's licensed under LGPL. Our project is a commercial one and from what I understand in order to use it, we have to disclosure our source code which is a big no no.

Any chance you can have this licensed under Apache? JavaCV has a wrapper for FFmpeg and is licensed under Apache, so I think this would be possible. 

This would save a lot of sweat, tears and keyboard if we can just simply use your lib :)

**Comment by wseemann** [comment]
FFmpegMediaMetadataRetriever library is licensed under the Apache License. FFmpeg is also statically linked in such a way as to allow it to be released as such.

**Comment by wseemann** [comment]
@shaileshmulange use the prebuilt libs with HTTPS support found [here](https://github.com/wseemann/FFmpegMediaMetadataRetriever/blob/master/prebuilt-aars-with-https.zip?raw=true).

**Comment by vxhviet** [comment]
@wseemann awesome, you are a life saver. Thanks a lot man. That's my bad actually, I thought the FFmpeg license part is FFmpegMediaMetadataRetriever's license.

**Comment by wseemann** [comment]
@vxhviet No problem, glad the library is useful!
